witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -167,11 +172,21 @@ func (w *Witness) Update(ctx context.Context, logID string, chkptRaw []byte, pro
 	if err != nil {
 		return 0, fmt.Errorf("couldn't create tx: %v", err)
 	}
-	latest, err := w.getLatestChkpt(tx.QueryRow, logID)
+	// Get either just the checkpoint or the checkpoint and compact range.
+	var latest Chkpt
+	var curRange [][]byte
+	if logInfo.UseCompact {
+		latest, curRange, err = w.getLatestChkptAndRange(tx.QueryRow, logID)
```

Comment:
Yeah, I knew this wasn't so nice :).  I've done this now, which turned into a slightly bigger change because I also took the opportunity to strip `size` from the database (it was already unnecessary since we haven't been storing multiple checkpoints for a while) and get rid of the `Chkpt` struct as a result too. 

Anyway both either take in or return raw `[]byte` arguments now and it's up to the caller to (un)marshal.